### PR TITLE
IntelliJ and Flutter plugin version checks

### LINKED ISSUE #1054: add a min version into the IntelliJ checks in flutter doctor
https://github.com/flutter/flutter/issues/8118

### PR DESCRIPTION
**Pull request by devoncarew**
- validate the installed versions of IntelliJ and the flutter plugin (fix https://github.com/flutter/flutter-intellij/issues/1054)
- instead of printing `version unknown` when we don't discover a version for android studio, just don't print a version
- some tweaks to our other version text for consistency

### CODE REVIEW

**Comment by devoncarew** [comment]
@tvolkert

File: packages/flutter_tools/lib/src/doctor.dart [hunk 1]
```diff
@@ -273,46 +278,73 @@ abstract class IntelliJValidator extends DoctorValidator {
...
         '$title plugin not installed; this adds $title specific functionality.'
       ));
-      return false;
+      return;
+    }
+    final String versionText = _readPackageVersion(packageName);
+    final Version version = new Version.parse(versionText);
+    if (version != null && minVersion != null && version < minVersion) {
+        messages.add(new ValidationMessage.error(
+          '$title plugin ${version != null ? "version $versionText" : "installed"}'
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/doctor.dart]: `version` is guaranteed to be non-null here.

File: packages/flutter_tools/lib/src/android/android_studio_validator.dart [hunk 2]
```diff
@@ -38,7 +38,9 @@ class AndroidStudioValidator extends DoctorValidator {
   Future<ValidationResult> validate() async {
     final List<ValidationMessage> messages = <ValidationMessage>[];
     ValidationType type = ValidationType.missing;
-    final String studioVersionText = 'version ${_studio.version}';
+    final String studioVersionText = Version.unknown == _studio.version
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/android/android_studio_validator.dart]: Nit: `_studio.version == Version.unknown`

**Review by tvolkert — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/android/android_studio_validator.dart (hunk 2 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/android/android_studio_validator.dart]: done

File: packages/flutter_tools/lib/src/doctor.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/doctor.dart]: updated

**Pull request merged**